hydroshare/hydroshare
# NetCDF resources migration to composite resource type

### Pull Request Checklist: 















### Positive Test Case
1. Testing locally:
    - Deploy the branch '3135-netcdf-resources-migration'. 
    - Edit the 'hs_core/templates/pages/create-resource.html' file to enable creating NetCDF resource.
    - Create couple of NetCDF  resources with a nc file upload using the local hydroshare website. You can use nc file (netcdf_valid.nc) that is in the hs_app_netCDF/tests folder for upload. Verify that each of the NetCDF resources you created has one nc file and one txt file. 
    - Run the management command 'migrate_netcdf_resources'. Output of the command should indicate the number of netcdf resources that were converted to composite resources and that there are 0 number of netcdf resources after the conversion. 
    - Click 'MY  RESOURCES' link on the webpage. You should see all resources are of type composite resource.
    - Go to the landing page of one of the composite resources. You should see that it contains one netcdf aggregation. Verify the metadata of this aggregation. Verify that the aggregation contains the original nc and txt files.
    - Download the bag for one of the composite resources. Verify the contents of the bag file that the expected resources files are there. Check that the 'resourcemap.xml' and 'resourcemetadata.xml' files do not contain the word 'NetCDFResource'.
2. Testing with real NetCDF resources (**this test is essential for approving this PR**):
    - Deploy branch '3135-netcdf-resources-migration' to deployment environment similar to 'beta' deployment where it would have access to a copy of real hydroshare resources.
    - Make a note of 'title' and 'resource id' of few NetCDF resources so that you will able to check those resources after they are converted to composite resource.
    - Run the management command 'migrate_netcdf_resources'. Output of the command should indicate the number of netcdf resources that were converted to composite resources and that there are 0 number of netcdf resources after the conversion. 
   - Use the web interface to check that there are no netcdf resources. Use your note to verify that those resources have been converted to composite resource. Verify that such a converted composite resource has one netcdf aggregation (containing one nc file and one txt file)  and you can view the aggregation level metadata. Check that you are able to download the bag file for one of these converted resources. Verify the contents of the bag file.
